components/server/src/user/user-service.ts

```diff
@@ -142,7 +152,8 @@ export class UserService {
                 this.config.blockNewUsers.passlist.some((e) => mail.endsWith(`@${e}`));
             const canPass = newUser.identities.some((i) => !!i.primaryEmail && emailDomainInPasslist(i.primaryEmail));
 
-            newUser.blocked = !canPass;
+            // blocked = if user already blocked OR is not allowed to pass
+            newUser.blocked = newUser.blocked || !canPass;
```

Comment:
![you-shall-not-pass](https://user-images.githubusercontent.com/599268/164029940-8198ed8f-d1b0-4d12-b8b5-2411d94c081b.jpg)